src/device/riscv/handleinterrupt.S

```diff
@@ -0,0 +1,130 @@
+#ifdef F_EXTENSION
```

Comment:
Same here, I think you can use `__riscv_flen`. See: https://github.com/riscv/riscv-toolchain-conventions